python/ray/serve/http_proxy.py

```diff
@@ -155,27 +157,55 @@ async def __call__(self, scope, receive, send):
             shard_key=headers.get("X-SERVE-SHARD-KEY".lower(), None),
         )
 
-        retries = 0
-        while retries <= MAX_ACTOR_DEAD_RETRIES:
-            try:
-                result = await self.router_handle.enqueue_request.remote(
-                    request_metadata, scope, http_body_bytes)
-                if not isinstance(result, ray.exceptions.RayActorError):
-                    await Response(result).send(scope, receive, send)
-                    break
-                logger.warning("Got RayActorError: {}".format(str(result)))
-                await asyncio.sleep(0.1)
-            except Exception as e:
-                error_message = "Internal Error. Traceback: {}.".format(e)
-                await error_sender(error_message, 500)
-                break
-        else:
-            logger.debug("Maximum actor death retries exceeded")
-            await error_sender(
-                "Internal Error. Maximum actor death retries exceeded", 500)
+        future = await self.router.enqueue_request(request_metadata, scope,
+                                                   http_body_bytes)
+        await Response(await future).send(scope, receive, send)
+
+
+def _autoproxy(*, methods, through):
+    """Decorator for a class. Automatically proxy methods
+
+    This decorator installs method with the same name in ``methods`` and proxy
+    them with the self.``through`` method.
+
+    Example:
+        >>> class A:
+                def call_one(self): pass
+                def call_two(self): pass
+        >>> @_autoproxy(methods=[A.call_one, A.call_two], through="_proxy")
+            class B:
+                a = A()
+                def _proxy(self, method_name, *args):
+                    return getattr(self.a, method_name)(*args)
+        >>> B.call_one()
+        >>> B.call_two()
+    """
+
+    def proxy_with_methods(cls):
+        for method in methods:
+            method_name = str(method.__name__)
+
+            async def wrapped_method(self, *args, name=method_name, **kwargs):
+                proxy_call = getattr(self, through)
+                return await proxy_call(name, *args, **kwargs)
+
+            setattr(cls, method_name, wrapped_method)
+        return cls
+
+    return proxy_with_methods
 
 
 @ray.remote
+@_autoproxy(
+    methods=[
+        Router.add_new_worker,
+        Router.set_traffic,
+        Router.set_backend_config,
+        Router.remove_backend,
+        Router.remove_endpoint,
+        Router.remove_worker,
+    ],
+    through="_proxy_router_call")
```

Comment:
This isn't really saving us much in terms of LOC and it adds complexity so I would prefer to just wrap the methods manually